xain_fl/grpc/test_grpc_updated.py

```diff
@@ -0,0 +1,247 @@
+# TODO: Once the old grpc participant is removed (XP-208), rename the module to `test_grpc`.
+
+import sys
+import threading
+from concurrent import futures
+from unittest import mock
+
+import grpc
+import numpy as np
+import pytest
+from numproto import ndarray_to_proto, proto_to_ndarray
+
+from xain_fl.grpc import (
+    coordinator_pb2,
+    coordinator_pb2_grpc,
+    hellonumproto_pb2,
+    hellonumproto_pb2_grpc,
+)
+from xain_fl.grpc.coordinator import (
+    Coordinator,
+    CoordinatorGrpc,
+    Participants,
+    monitor_heartbeats,
+)
+from xain_fl.grpc.participant import (
+    StateRecord,
+    end_training,
+    message_loop,
+    rendezvous,
+    start_training,
+)
+
+# Some grpc tests fail on macos.
+# `pytestmark` when defined on a module will mark all tests in that module.
+# For more information check
+# http://doc.pytest.org/en/latest/skipping.html#skip-all-test-functions-of-a-class-or-module
+if sys.platform == "darwin":
+    pytestmark = pytest.mark.xfail(reason="some grpc tests fail on macos")
+
+# pylint: disable=W0613,W0621
+@pytest.mark.integration
+def test_greeter_server(greeter_server):
+    with grpc.insecure_channel("localhost:50051") as channel:
+        stub = hellonumproto_pb2_grpc.NumProtoServerStub(channel)
+
+        nda = np.arange(10)
+        response = stub.SayHelloNumProto(
+            hellonumproto_pb2.NumProtoRequest(arr=ndarray_to_proto(nda))
+        )
+
+        response_nda = proto_to_ndarray(response.arr)
+
+        assert np.array_equal(nda * 2, response_nda)
+
+
+@pytest.mark.integration
+def test_participant_rendezvous_accept(participant_stub, coordinator_service):
+    reply = participant_stub.Rendezvous(coordinator_pb2.RendezvousRequest())
+
+    assert reply.response == coordinator_pb2.RendezvousResponse.ACCEPT
+
+
+# TODO: Fix test so it also runs correctly on macos
```

Comment:
I think there is already a ticket for this TODO, right?

Reply:
Added ticket number XP-208.